sysctl will now return an error if the value is invalid

kustodian opened the pull request:
##### SUMMARY
sysctl can fail to set a value even if it returns an exit status 0. More details: https://bugzilla.redhat.com/show_bug.cgi?id=1264080. Because of this in case of an invalid value or a read-only file system, sysctl module would return OK, even though it didn't set anything. To be sure that sysctl correctly applied the changes we also need to check the output of stderr.

What I'm still not sure if we should return as failed if there is anything in stderr?

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
sysctl

kustodian committed:
sysctl will now return an error if the value is invalid

sysctl can fail to set a value even if it returns an exit status 0. More
details: https://bugzilla.redhat.com/show_bug.cgi?id=1264080. Because of
this in case of an invalid value or a read-only file system, sysctl
module would return OK, even though it didn't set anything. To be sure
that sysctl correctly applied the changes we also need to check the
output of stderr.

kustodian commented:
While working on this fix, I also noticed that there is one more problematic behavior of this module. If you use `sysctl_set=no` (which is the default) and if the value in the `/etc/sysctl.conf` is different than the current value on the system, the module won't report as changed and it won't do anything. IMO the module should always check both the value in the file and the current value set on the system. If I set some value with this module, I want that value to be the current value on the system. If for some reason I don't want that, then I will use `reload=no`.

bcoca reviewed (approved):
@kustodian I also agree with your assessment on how check mode should work. Feel free to open a bug report so we can keep track that we need to fix that behaviour.

kustodian commented:
@bcoca I'll make a PR with the fix for check mode, because it should be an easy fix.

samdoran commented:
Please create a changelog fragment. See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

kustodian committed:
Run sysctl with LANG=C

Because we are parsing sysctl stderr we need to make sure that errors
are persistent across different system language settings.

kustodian committed:
Add changelog fragment for sysctl

kustodian commented:
I wasn't around last week, but here are all the needed fixes.

@bcoca you suggested on IRC that we should make sure that `sysctl` is run with `LANG=C`, because we are parsing STDERR. This totally makes sense, so I'm adding that as well.

@samdoran Fragment is added. I always forget to add this fragment. Maybe you guys could somehow add a test for this so that all module PRs need to have a fragment?

samdoran commented:
Please create a [backport PR](https://docs.ansible.com/ansible/latest/community/development_process.html#backport-pull-request-process) for this to be included in previous versions.
